### PR TITLE
HIVE-25904: ObjectStore's updateTableColumnStatistics is not ThreadSafe

### PR DESCRIPTION
**Pull request by deniskuzZ**

### CODE REVIEW

File: standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java [hunk 1]
```diff
@@ -391,6 +395,15 @@ public void setConf(Configuration conf) {
     } else {
       LOG.debug("Initialized ObjectStore");
     }
+
+    if (tablelocks == null) {
+      synchronized (ObjectStore.class) {
+        if (tablelocks == null) {
+          int numTableLocks = MetastoreConf.getIntVar(conf, ConfVars.METASTORE_NUM_STRIPED_TABLE_LOCKS);
+          tablelocks = Striped.lock(numTableLocks);
```
Comment (rbalamohan) [inline standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java:403]: Is this different from ReentrantLock? (PaddingLock extends ReentrantLock and adds padding for cache line). But not heavily benchmarked. E.g https://github.com/google/guava/issues/3195

Changing to ReenetrantLock may be easier to read?

Comment (deniskuzZ) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java:403]: sure, @rbalamohan thanks for taking a look at this!

Comment (deniskuzZ) [reply standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java]: @rbalamohan, are you ok with the change?

**Comment by rbalamohan** [comment]
LGTM. +1.

**Pull request merged**